lambdas/issuecredential/src/main/java/uk/gov/di/ipv/cri/passport/issuecredential/handler/IssueCredentialHandler.java

```diff
@@ -0,0 +1,211 @@
+            APIGatewayProxyRequestEvent input, Context context) {
+
+        try {
+            LOGGER.info(
+                    "Initiating lambda {} version {}",
+                    context.getFunctionName(),
+                    context.getFunctionVersion());
+
+            LOGGER.info("Validating authorization token...");
+            var accessToken = validateInputHeaderBearerToken(input.getHeaders());
+            var sessionItem = this.sessionService.getSessionByAccessToken(accessToken);
+            LOGGER.info("Extracted session from session store ID {}", sessionItem.getSessionId());
+
+            LOGGER.info("Retrieving identity details and document check results...");
+            var personIdentityDetailed =
+                    personIdentityService.getPersonIdentityDetailed(sessionItem.getSessionId());
+            DocumentCheckResultItem documentCheckResult =
+                    documentCheckResultStore.getItem(sessionItem.getSessionId().toString());
+
+            if (documentCheckResult == null) {
+                LOGGER.error("User has arrived in issue credential without completing check");
+                return ApiGatewayResponseGenerator.proxyJsonResponse(
+                        HttpStatusCode.INTERNAL_SERVER_ERROR,
+                        OauthErrorResponse.ACCESS_DENIED_ERROR);
+            }
+
+            LOGGER.info("VC content retrieved.");
+
+            LOGGER.info("Generating verifiable credential...");
+            SignedJWT signedJWT =
+                    verifiableCredentialService.generateSignedVerifiableCredentialJwt(
+                            sessionItem.getSubject(), documentCheckResult, personIdentityDetailed);
+            LOGGER.info("Credential generated");
+
+            String verifiableCredentialIssuer =
+                    passportConfigurationService.getVerifiableCredentialIssuer();
+
+            auditService.sendAuditEvent(
+                    AuditEventType.VC_ISSUED,
+                    new AuditEventContext(input.getHeaders(), sessionItem),
+                    IssueCredentialPassportAuditExtensionUtil
+                            .generateVCISSDocumentCheckAuditExtension(
+                                    verifiableCredentialIssuer, List.of(documentCheckResult)));
+
+            // CI Metric captured here as check lambda can have multiple attempts
+            recordCIMetrics(PASSPORT_CI_PREFIX, documentCheckResult.getContraIndicators());
+
+            auditService.sendAuditEvent(
+                    AuditEventType.END, new AuditEventContext(input.getHeaders(), sessionItem));
+
+            eventProbe.counterMetric(LAMBDA_ISSUE_CREDENTIAL_COMPLETED_OK);
+
+            return ApiGatewayResponseGenerator.proxyJwtResponse(
+                    HttpStatusCode.OK, signedJWT.serialize());
+        } catch (AwsServiceException ex) {
+            LOGGER.error(LAMBDA_EXCEPTION_ERROR_MESSAGE, context.getFunctionName(), ex.getClass());
```

Comment:
We should also leave a comment here to restore the stack logging asap or we should log stacks from any were confident wont contain PII